specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json

```diff
@@ -0,0 +1,975 @@
+                        "required": true
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "client_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD client id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with object_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "object_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD object id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with client_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "mi_res_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by urlencoded ARM resource id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with client_id and object_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "authority",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This indicates the authority to request AAD tokens from. Defaults to the known authority of the identity to be used.",
+                        "required": false
+                    },
+                    {
+                        "name": "bypass_cache",
+                        "in": "query",
+                        "type": "string",
+                        "enum": [
+                            "true"
+                        ],
+                        "description": "If provided, the value must be 'true'. This indicates to the server that the token must be retrieved from Azure AD and cannot be retrieved from an internal cache.",
+                        "required": false,
+                        "x-ms-enum": {
+                            "name": "BypassCache",
+                            "modelAsString": true
+                        }
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Successful operation",
+                        "schema": {
+                            "$ref": "#/definitions/IdentityTokenResponse"
+                        }
+                    },
+                    "401": {
```

Comment:
From an API perspective, I have no concerns. However, if this document is to be used for autorest, adding additional failure cases in the responses section will negatively impact the generated code.

Reply:
I removed 401 and added headers to default section. As headers is optional, so we won't break the existing scenarios. Thanks @johanste.